core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -996,6 +1005,13 @@ private[spark] class SparkSubmit extends Logging {
           SparkContext.getActive.foreach(_.stop())
         } catch {
           case e: Throwable => logError(s"Failed to close SparkContext: $e")
+        } finally {
+          if (SparkContext.getActive.isEmpty) {
+            if (!DriverPodIsNormal) {
+              logError(s"Driver Pod will exit because: $driverThrow")
+              System.exit(EXIT_FAILURE)
```

Comment:
Could you provide the YARN code link for this case? Specifically,
1. Does YARN job fail with EXIT_FAILURE ?
2. If this PR is for consistency between YARN and K8s,  we should have the same exit code and same error message.

Reply:
Here is yarn exit code.
[resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala](https://github.com/apache/spark/blob/39500a315166d8e342b678ef3038995a03ce84d6/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala#L870)
It seems applicationmaster adopts it's own failure exit code, which not suitable for driver pod.

Reply:
What is the YARN exit code for your case? According to your PR description, it's not clear, @littlelittlewhite09 . It would be described in the PR description.

> When spark app runs on yarn-cluster mode, everything is ok. Driver can terminate normally if encounters exception or err.